### PR TITLE
archlinux: improve pacman proxy implementation

### PR DESCRIPTION
**Pull request by ejose19**

### CODE REVIEW

**Comment by ejose19** [comment]
@fepitre Let me know what you think

**Comment by fepitre** [comment]
@ejose19 It looks good to me. I would need to test it too.

File: network/update-proxy-configs [hunk 1]
```diff
@@ -119,21 +119,26 @@ fi
 # Pacman (archlinux) also
 if [ -d /etc/pacman.d ]; then
     if [ -n "$PROXY_ADDR" ]; then
-        cat > /etc/pacman.d/01-qubes-proxy.conf <<EOF
+        mkdir -p /usr/lib/qubes/bin
+        cat > /usr/lib/qubes/bin/pacman <<EOF
```
Comment (marmarek) [inline network/update-proxy-configs:123]: If going with a non-standard path anyway, can we please not dynamically modify content of `/usr`? Better put it in `/var` or even `/run`

File: network/update-proxy-configs [hunk 2]
```diff
@@ -119,21 +119,26 @@ fi
...
+        chmod +x /usr/lib/qubes/bin/pacman
+        cat > /etc/profile.d/qubes-proxy.sh << EOF
+### This file is automatically generated by Qubes ($0 script).
+### All modifications here will be lost.
+export PATH=/usr/lib/qubes/bin:\$PATH
 EOF
     else
-        rm -r /etc/pacman.d/01-qubes-proxy.conf
+        rm -r /usr/lib/qubes/bin/pacman
+        rm -r /etc/profile.d/qubes-proxy.sh
```
Comment (marmarek) [inline network/update-proxy-configs:137]: Both wants a `-f` instead of `-r`.

File: network/update-proxy-configs [hunk 3]
```diff
@@ -119,21 +119,26 @@ fi
...
+#!/bin/bash
 ### This file is automatically generated by Qubes ($0 script).
 ### All modifications here will be lost.
-### If you want to override some of this settings, create another file under 
-### /etc/pacman.d
-
-[options]
-# Use Qubes Update Proxy
-XferCommand = /usr/bin/env ALL_PROXY=$PROXY_ADDR /usr/bin/curl -C - -f -o %o %u
+ALL_PROXY=$PROXY_ADDR /usr/bin/pacman \$@
```
Comment (marmarek) [inline network/update-proxy-configs:127]: Add `exec` there, to avoid extra process.

**Review by marmarek — CHANGES_REQUESTED** [review]

**Comment by ejose19** [comment]
@marmarek Should be good now. 

P.S. No idea why the code-signing check still fails, I can run ```gpg --keyserver pool.sks-keyservers.net --recv-key 09BEDF7CC137762C``` on a dispvm and it returns the key.

File: network/update-proxy-configs [hunk 4]
```diff
@@ -119,21 +119,26 @@ fi
...
+#!/bin/bash
 ### This file is automatically generated by Qubes ($0 script).
 ### All modifications here will be lost.
-### If you want to override some of this settings, create another file under 
-### /etc/pacman.d
-
-[options]
-# Use Qubes Update Proxy
-XferCommand = /usr/bin/env ALL_PROXY=$PROXY_ADDR /usr/bin/curl -C - -f -o %o %u
+exec env ALL_PROXY=$PROXY_ADDR /usr/bin/pacman \$@
```
Comment (marmarek) [inline network/update-proxy-configs:127]: ah, one more thing: it should be `"$@"`, otherwise arguments with spaces won't be handled correctly.

Comment (ejose19) [reply network/update-proxy-configs:127]: Can you give me an example of such command so next time I can do a better testing? I tried with ```pacman -S -y -u --color always``` and it worked without issues.

Comment (marmarek) [reply network/update-proxy-configs]: It's about arguments with spaces inside, for example:
`pacman --print --print-format "package %n version %v" -U`
(not sure if I got the syntax right, written based on pacman man page)

**Review by marmarek — APPROVED** [review]

File: network/update-proxy-configs (hunk 4 above)
Comment (ejose19) [reply network/update-proxy-configs]: I tried your command and it failed without the double quotes, thanks.

**Comment by ejose19** [comment]
https://github.com/QubesOS/qubes-issues/issues/5585 Can be closed now as this is merged

**Pull request merged**